proxies/live/apiproxy/resources/jsc/SetErrorVars.js

```diff
@@ -0,0 +1,200 @@
+          } else if (message === null) {
+            errors.push(nullError(pointer))
+          } else {
+
+          // $.data.attributes.messages.x.messageReference
+          pointer = "/data/attributes/messages/" + index + "/messageReference"
+          if (typeof message.messageReference === 'undefined') {
+            errors.push(missingError(pointer))
+          } else if (message.messageReference === null) {
+            errors.push(nullError(pointer))
+          } else if (typeof message.messageReference !== "string" || !uuidRegex.test(message.messageReference)) {
+            errors.push(invalidError(pointer))
+          } else if (message.messageReference in seenMessages) {
+            errors.push(duplicateError(pointer))
+          }
+          seenMessages.push(message.messageReference)
+          }
+
+          // $.data.attributes.messages.x.recipient
+          pointer = "/data/attributes/messages" + index + "/recipient"
+          if (typeof message.recipient === 'undefined') {
+            errors.push(missingError(pointer))
+          } else if (message.recipient === null) {
+            errors.push(nullError(pointer))
+          }
+
+          // $.data.attributes.messages.x.recipient.nhsNumber
+          pointer = "/data/attributes/recipients/" + i + "/recipient/nhsNumber"
+          if (typeof message.recipient.nhsNumber === 'undefined') {
+            errors.push(missingError(pointer))
+          } else if (message.recipient.nhsNumber === null) {
+            errors.push(nullError(pointer))
+          } else if (typeof message.recipient.nhsNumber !== "string" || !nhsNumberRegex.test(message.recipient.nhsNumber)) {
+            errors.push(invalidError(pointer))
+          }
+
+          // $.data.attributes.recipients.x.dateOfBirth
+          pointer = "/data/attributes/recipients/" + index + "/recipient/dateOfBirth"
+          if (typeof message.recipient.dateOfBirth === 'undefined') {
+            errors.push(missingError(pointer))
+          } else if (message.recipient.dateOfBirth === null) {
+            errors.push(nullError(pointer))
+          } else if (typeof message.recipient.dateOfBirth !== "string" || !dobRegex.test(message.recipient.dateOfBirth)) {
+            errors.push(invalidError(pointer))
+          }
+
+          // $.data.attributes.messages.x.personalisation
+          pointer = "/data/attributes/messages/" + index + "/personalisation"
+          if (typeof message.personalisation === 'undefined') {
+            errors.push(missingError(pointer))
+          }
+
+        })
+      }
+    }
+}
+
+// function findProperty(obj, property, defval){
+//     if (typeof defval == 'undefined') defval = null;
+//     property = property.split('/');
+//     for (var i = 0; i < property.length; i++) {
+//         if(typeof obj[property[i]] == 'undefined')
+//             return defval;
+//         obj = obj[property[i]];
+//     }
+//     return obj;
+// }
+
+function createErrorObject(title, detail, pointer) {
```

Comment:
I'll probably move all of these error message generating functions into an external library file for the sake of conciseness